### PR TITLE
Keep CommandMenu dismissed after Escape

### LINKED ISSUE #930: CommandMenu: Escape leaves the / trigger in the input, which immediately reopens the menu
## Symptom
Pressing Escape inside CommandMenu closes the menu, but because the `/` trigger character remains in the input, the menu reopens immediately — Escape is effectively inert from the user's perspective.

## Root cause
`command-menu.svelte:147` wires `onEscape: dismiss`. `dismiss` closes the menu but does not remove or neutralize the `/` trigger character in the input. The menu opens by detecting that trigger via `detectTrigger` (exported `command-menu-trigger.ts:21`), so with the caret still sitting at/after a `/`, the open condition is immediately satisfied again.

**The existing tests miss exactly this case.** `command-menu.test.ts:178` (`'Escape dismisses the menu'`, Escape fired at `:188`) and `command-menu.test.ts:336` both assert the menu is **closed** immediately after Escape — never that it **stays** closed on a subsequent tick/re-evaluation. Fixing the escape bug will not be caught or prevented by current coverage.

## Fix — requires a state decision, not just calling dismiss()
Does Escape clear the `/` from the input, or leave the text and suppress re-triggering until some other action clears the suppression (keystroke, caret move, blur, explicit re-invocation of the trigger)? Enumerate the edge-case matrix before implementing:
- Escape with a query typed after the `/` (e.g., `/foo`)
- Escape then immediately retyping `/`
- Escape with an option highlighted
- Blur-then-refocus while `/` is still present in the input
- Undo/redo restoring a `/` into the input

## Acceptance criteria
- Pressing Escape closes the menu and it **stays closed** — it must not reopen on the same trigger state without a new, deliberate user action.
- The decision made for each edge case above is implemented consistently and documented in a comment near `detectTrigger`/`dismiss`.

## Regression test
Update or add a test that asserts the menu remains closed after Escape across at least: bare `/`, `/` with a query, and blur/refocus with `/` still present — not just that it's closed in the tick immediately following the Escape keypress.

### PR DESCRIPTION
**Pull request by stevekinney**
## Summary

- remember the anchor value and caret state dismissed by Escape
- reject host-triggered reopen attempts while that exact trigger state is unchanged
- clear the guard on deliberate input, navigation, pointer interaction, paste, or undo/redo
- cover bare slash, slash queries, Escape keyup, blur/refocus, and restored input state

## Root cause

CommandMenu handled Escape on `keydown`, but host examples re-ran `detectTrigger` on `keyup`. Because Escape left the host-owned slash text unchanged, the keyup handler immediately set `open` back to true.

## Verification

- `bun test packages/components/src/components/command-menu/command-menu.test.ts`
- `bun run --filter=@lostgradient/cinder typecheck`
- `bun run --filter=@lostgradient/cinder components:check`
- `bun run --filter=@lostgradient/cinder exports:check`
- `bun run --filter=@lostgradient/cinder check:changeset-prerelease-bumps`
- targeted Prettier and Oxlint checks

Closes #930

### CODE REVIEW

File: packages/components/src/components/command-menu/command-menu.svelte [hunk 1]
```diff
@@ -154,9 +190,29 @@
...
   }
 
+  $effect(() => {
+    if (!anchor) return;
+
+    const clearEscapeDismissal = () => {
+      escapeDismissal = null;
+    };
+    const clearEscapeDismissalOnKeydown = (event: KeyboardEvent) => {
+      if (event.key !== 'Escape') clearEscapeDismissal();
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/command-menu/command-menu.svelte]: Fixed in 288e4c52. The guard now clears only on input or explicit pointer interaction; unchanged modifier keydowns remain dismissed. Added Shift keydown/keyup coverage to the bare-slash and query cases.

File: packages/components/src/components/command-menu/command-menu.svelte [hunk 2]
```diff
@@ -154,9 +190,24 @@
...
   }
 
+  $effect(() => {
+    if (!anchor) return;
+
+    const clearEscapeDismissal = () => {
+      escapeDismissal = null;
+    };
+    const stopInput = on(anchor, 'input', clearEscapeDismissal, { capture: true });
+    const stopPointerdown = on(anchor, 'pointerdown', clearEscapeDismissal, { capture: true });
```
[the remark this answers is not in the document]
Comment (stevekinney) [reply packages/components/src/components/command-menu/command-menu.svelte:214]: Fixed in 5d0454d9: the closed menu now observes keyup/select changes on its anchor and clears the Escape guard only when the caret range differs. Regression covers left-away/right-back reopening; focused suite is 22/22.
